src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -265,4 +267,15 @@ private function getReflectionClass($id, Definition $definition)
             // return null
         }
     }
+
+    private function addServiceToAmbiguousType($id, $type)
+    {
+        // keep an array of all services matching this type
+        if (!isset($this->ambiguousServiceTypes[$type])) {
+            $this->ambiguousServiceTypes[$type] = array(
+                $this->types[$type],
```

Comment:
In that case, the call to ` $this->types[$type]` is always skipped because of `if (!isset($this->ambiguousServiceTypes[$type])) {` (the line you pointed is in a `isset($this->ambiguousServiceTypes[$type])` block.